latch_cli/snakemake/workflow.py

```diff
@@ -0,0 +1,490 @@
+            )
+        params[k] = interface_models.Parameter(
+            var=v, default=default_lv, required=required
+        )
+    return interface_models.ParameterMap(params)
+
+
+class SnakemakeWorkflow(WorkflowBase, ClassStorageTaskResolver):
+    def __init__(
+        self,
+        name: str,
+        dag: DAG,
+    ):
+
+        parameter_metadata = {}
+        for job in dag.jobs:
+            for x in job.input:
+                var = variable_name_for_value(x, job.input)
+                parameter_metadata[var] = LatchParameter(display_name=var)
+
+        # TODO - support for metadata + parameters in future releases
+        latch_metadata = LatchMetadata(
+            display_name=name,
+            documentation="",
+            author=LatchAuthor(
+                name="",
+                email="",
+                github="",
+            ),
+            parameters=parameter_metadata,
+            tags=[],
+        )
+        docstring = Docstring(f"{name}\n\nSample Description\n\n" + str(latch_metadata))
+
+        native_interface = snakemake_dag_to_interface(dag, docstring)
+
+        self._input_parameters = None
+        self._dag = dag
+        self.snakemake_tasks = []
+
+        workflow_metadata = WorkflowMetadata(
+            on_failure=WorkflowFailurePolicy.FAIL_IMMEDIATELY
+        )
+        workflow_metadata_defaults = WorkflowMetadataDefaults(False)
+        super().__init__(
+            name=name,
+            workflow_metadata=workflow_metadata,
+            workflow_metadata_defaults=workflow_metadata_defaults,
+            python_interface=native_interface,
+        )
+
+    def compile(self, **kwargs):
+
+        self._input_parameters = interface_to_parameters(self.python_interface)
+
+        GLOBAL_START_NODE = Node(
+            id=_common_constants.GLOBAL_INPUT_NODE_ID,
+            metadata=None,
+            bindings=[],
+            upstream_nodes=[],
+            flyte_entity=None,
+        )
+
+        node_map = {}
+
+        target_files = [x for job in self._dag.targetjobs for x in job.input]
+
+        for layer in self._dag.toposorted():
+            for job in layer:
+
+                is_target = False
+
+                if job in self._dag.targetjobs:
+                    continue
+
+                target_file_for_output_param: Dict[str, str] = {}
+                target_file_for_input_param: Dict[str, str] = {}
+
+                python_outputs: Dict[str, Type] = {}
+                for x in job.output:
+                    if x in target_files:
+                        is_target = True
+                    param = variable_name_for_value(x, job.output)
+                    target_file_for_output_param[param] = x
+                    python_outputs[param] = LatchFile
+
+                @dataclass
+                class JobOutputInfo:
+                    jobid: str
+                    output_param_name: str
```

Comment:
define this outside of the method

Reply:
also add `frozen=True`

Reply:
why do i care if people modify this value, i don't really see the point